pennylane/templates/state_preparations.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+State Preparations
+==========
+
+**Module name:** :mod:`pennylane.templates.state_preparations`
+
+.. currentmodule:: pennylane.templates.state_preparations
+
+This module provides routines that prepare a given state using only
+elementary gates.
+
+Qubit architectures
+-------------------
+
+.. autosummary::
+
+    BasisStatePreparation
+    MottonenStatePreparation
+
+Code details
+^^^^^^^^^^^^
+"""
+from collections.abc import Iterable
+import numpy as np
+
+import pennylane as qml
+
+
+def BasisStatePreparation(basis_state, wires):
+    r"""
+    Prepares a basis state on the given wires using a sequence of Pauli X gates.
+
+    Args:
+        basis_state (array): Input array of shape ``(N,)``, where N is the number of qubits,
+            with :math:`N\leq n`
```

Comment:
what is `n` in this case?

Reply:
Number of wires. I copied this from the other templates. I'll change it in both places.